yjit/src/codegen.rs

```diff
@@ -4133,6 +4153,87 @@ fn gen_return_branch(
     }
 }
 
+
+fn pop_args_splat(argc: i32, ctx: &mut Context, asm: &mut Assembler, ocb: &mut OutlinedCb, side_exit: CodePtr, num_params: u32) -> i32 {
+
+    let mut argc = argc;
+
+    asm.comment("pop_args_splat 4");
+
+    let array_type = ctx.get_opnd_type(StackOpnd(argc as u16 - 1));
+    let array_opnd = ctx.stack_opnd(0);
+
+    argc = argc - 1;
+
+    // Move the array from the stack into REG0 and check that it's an array.
+    let array_reg = asm.load(array_opnd);
+    guard_object_is_heap(
+        asm,
+        array_reg,
+        counted_exit!(ocb, side_exit, send_splat_not_array),
+    );
+    guard_object_is_array(
+        asm,
+        array_reg,
+        counted_exit!(ocb, side_exit, send_splat_not_array),
+    );
+
+    // Pull out the embed flag to check if it's an embedded array.
+    let flags_opnd = Opnd::mem((8 * SIZEOF_VALUE) as u8, array_reg, RUBY_OFFSET_RBASIC_FLAGS);
+
+    // Move the length of the embedded array into REG1.
```

Comment:
These comments need to be updated because there's no longer a REG0/REG1 with the new backend